src/services/milestones/milestones.hooks.js

```diff
@@ -201,6 +203,60 @@ const sendNotification = () => context => {
   }
 };
 
+/***
+ * This function checks that the maxAmount in the milestone is based on the correct conversion rate of the milestone date
+ **/
+const checkEthConversion = () => (context) => {
+  // abort check for internal calls
+  if (!context.params.provider) return context;
+
+  const { data, app } = context;
+  let maxAmount = 0;
+  const items = data.items;
+
+  const calculateCorrectEther = (conversionRate, fiatAmount, etherToCheck, selectedFiatType) => {
+    logger.info('calculating correct ether conversion', conversionRate.rates[selectedFiatType], fiatAmount, etherToCheck);
+    // calculate the converion of the item, make sure that fiat-eth is correct
+    const rate = conversionRate.rates[selectedFiatType];
+    const ether = utils.toWei((fiatAmount / rate).toFixed(18));
+
+    if (ether !== etherToCheck) {
+      throw new errors.Forbidden('Conversion rate is incorrect');
+    }    
+  }
+
+  if(items && items.length > 0) {
+    // check total amount of milestone, make sure it is correct
+    let totalItemEtherAmount = 0;
+    items.forEach((item) => totalItemEtherAmount += parseFloat(item.etherAmount))
+
+    if (utils.toWei(totalItemEtherAmount.toFixed(18)) !== data.maxAmount) {
+      throw new errors.Forbidden('Total amount in ether is incorrect');
+    } 
+
+    // now check that the conversion rate for each milestone is correct
+    return new Promise((resolve, reject) =>
+      items.forEach((item, index) => {
+        app.service('ethconversion')
+          .find({ query: { date: item.date }, internal: true })
+          .then((conversionRate) => {
+            calculateCorrectEther(conversionRate, item.fiatAmount, utils.toWei(item.etherAmount.toFixed(18)), item.selectedFiatType);
+            
+            if(index === 0) resolve(context);
```

Comment:
I think this may be a race condition. Since app.service is async, I think it's possible for index === 0 before all items have been checked.

I think it would be safer to use `Promise.all`

Reply:
hmmm good point. Will change